pandas/tests/io/formats/test_to_html.py

```diff
@@ -1881,3 +1881,40 @@ def test_to_html_with_index_names_false(self):
                                                         name='myindexname'))
         result = df.to_html(index_names=False)
         assert 'myindexname' not in result
+
+    def test_to_html_styler(self):
+        df = pd.DataFrame({"A": [-1, 1], "B": [-2, 2]}).style.applymap(
+            lambda x: 'color: {}'.format(
+                'red' if x < 0 else 'black'))
+        df.uuid = 42
+        expected = (
```

Comment:
Let's think about what we're testing here. I think we want to verify that the *writing* happens correctly. We can assuming that the styling is correct.

So, let's do something like

```python
result = df.to_html()
expected = df.render()
assert result == expected
```

instead of hard-coding the expected.

And we'll need to cover a few more cases:

1. writing to a file specified by a str
2. writing to a file specified by a pathlike object (pathlib.Path, io.StringIO)
3. Writing in append mode
4. Writing with a different encoding

For those tests, we can write the data, read it back in, and assert that the data read in matches `df.render()`. The append mode test would be a bit different (maybe write it once with `mode='w'` and then write again with `mode='a'` and assert the result is `df.render() + df.render()`) .

Reply:
I went ahead and changed the test -- my thinking had been if the expected result was hard-coded then it did not depend upon the behavior of `.render()`. In principle if the current test fails one cannot be sure which method actually failed. However, there's other places that use `.render()`, so maybe it is okay